db_stress_tool/db_stress_test_base.h

```diff
@@ -78,7 +78,10 @@ class StressTest {
     // first write fails
     if (!write_s.ok() && (*initial_write_s).ok()) {
       *initial_write_s = write_s;
+      // With commit_bypass_memtable_one_in > 0, error status may come from
+      // SwitchMemTable() which happens after WAL write.
       *initial_wal_write_may_succeed =
```

Comment:
The `initial_wal_write_may_succeed` set to true is used to retry the write until it succeeds in case the data is persisted to WAL but the whole write failed,  so that WAL doesn't diverge from expected state when a crash happen here, right?

If this write failure is in commit_bypass_memtable mode, and it's a failed to write to wal error, do we need to keep retrying until success, since above diverging concern is not a possibility.  Especially given that `FLAGS_commit_bypass_memtable_one_in` being bigger than 1 doesn't necessarily mean this transaction definitely used this mode.

Reply:
> If this write failure is in commit_bypass_memtable mode, and it's a failed to write to wal error, do we need to keep retrying until success, since above diverging concern is not a possibility.

With commit_bypass_memtable, there can be a WAL error from creating WAL SwitchMemtable() that happens after the write to WAL. In that case, data is persisted in WAL and we should retry the write. 

> FLAGS_commit_bypass_memtable_one_in being bigger than 1 doesn't necessarily mean this transaction definitely used this mode.

Yeah, let me update this part to reflect the actual transaction option.

Reply:
Thanks for the explanation. It's a bit tricky that the order of these two conditions actually matter even if it's a `||` operation. Maybe add some documentation for not changing orders or refactor to enforce that?

Reply:
Added some more explanation in comment and updated this part to check the actual transaction option.